### ISSUE TITLE
Squids spawning

### ISSUE THREAD

**Issue by x1p** [PaperMC/Paper#1692]
### What behaviour is observed:
In my world we've noticed a bit low squid numbers.

Bukkit.yml
```
spawn-limits:
  monsters: 75
  animals: 15
  water-animals: 15
  ambient: 3
```
I have the feeling that players with guardian farms eat up all the resources for spawning water animals. So if a few players are near their sea farms, other fish like animals don't spawn easy naturally anymore. I could use a mob limiter to kill guardians, but I don't think that is the solution. Raising the water-animals would probably increase the number of guardians, rather than divide the number of mobs between the online players.

### Steps/models to reproduce:
not easy to test

### Paper build number:
git-Paper-447 (MC: 1.13.2)

**Comment by electronicboy** [comment]
guardians have nothing to do with water mob limits, depending on your ranges, you may need to increase the value; There are also known issues in craftbukkit/vanilla due to some mechanical changes I've not been able to follow which causes issues with entity persistence which affects mob spawn limits

**Comment by x1p** [comment]
Would be nice if one can decide what mob limits to use per entity to alter the game for custom numbers.... wishful thinking...

**Comment by mart-r** [comment]
I believe the issue is with the new 1.13 fish and dolphins taking up the cap. It seems that there's dozens of dolphins around my AFK area every time I stop my AFK session.
But someone who actually knows can correct me if I'm wrong, of course.

**Comment by electronicboy** [comment]
Spigot ended up messing around with fish in 1.13, trying to get a clearer picture of why they went the route they did before I start poking at stuff too much

**Comment by StrangeDOS** [comment]
I'm experiencing this issue as well.  I have a squid farm of about 400 water columns.  If I use Paper, the number of squids that spawn drags.  The most that will spawn at once is 4...and they don't spawn very often.  However, if I switch the world back to vanilla, it spawns like crazy...17 at a time and constantly spawning.  Switch it back to Paper, and I'm back to just a few spawning.  

I'd gladly post my world if you need it for testing.

Minecraft version 1.13.2

**Comment by josephrooks** [comment]
Don't know why, but this seems to be fixed in 1.14.4. Squid spawns in my previously barely functional squid farm are now through the roof the way they would be in vanilla.

**Comment by Chew** [comment]
Considering this bug is ancient, and has had at least one "seems fixed" as of 1.14.4, it's time to go ahead and close this issue. In fact, we might have the opposite problem in #3523! If you are still experiencing this issue on latest versions, feel free to report a new issue. Thank you for the report!

**Comment by kalleswe** [comment]
And again broken in 1.18.1, with no squid spawning on default settings.
